From 25bfbbc0104e0c7e81ef54255f06455f24faceec Mon Sep 17 00:00:00 2001
From: Alexandre Julliard <julliard@winehq.org>
Date: Thu, 5 Feb 2026 10:55:31 +0100
Subject: [PATCH] wow64: Add missing Unicode string mapping for
 MemoryWineLoadUnixLibByName.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=59373
(cherry picked from commit 9cbdc12cdd3dcadaeff7122a0fb928be0a48e0d9)

CW-Bug-Id: #27687
---
 dlls/wow64/virtual.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/dlls/wow64/virtual.c b/dlls/wow64/virtual.c
index 1c2cdf19e4a..b4778a86efc 100644
--- a/dlls/wow64/virtual.c
+++ b/dlls/wow64/virtual.c
@@ -701,8 +701,14 @@ NTSTATUS WINAPI wow64_NtQueryVirtualMemory( UINT *args )
         status = NtQueryVirtualMemory( handle, addr, MemoryWineLoadUnixLibWow64, ptr, len, &res_len );
         break;
     case MemoryWineLoadUnixLibByName:
-        status = NtQueryVirtualMemory( handle, addr, MemoryWineLoadUnixLibByNameWow64, ptr, len, &res_len );
+    {
+        UNICODE_STRING32 *str32 = addr;
+        UNICODE_STRING str;
+
+        status = NtQueryVirtualMemory( handle, unicode_str_32to64( &str, str32 ),
+                                       MemoryWineLoadUnixLibByNameWow64, ptr, len, &res_len );
         break;
+    }
     case MemoryWineUnloadUnixLib:
         status = NtQueryVirtualMemory( handle, addr, class, ptr, len, &res_len );
         break;
